src/Core/Services/MetadataProviders/CosmosSqlMetadataProvider.cs

```diff
@@ -55,12 +56,14 @@ public class CosmosSqlMetadataProvider : ISqlMetadataProvider
 
         public CosmosSqlMetadataProvider(RuntimeConfigProvider runtimeConfigProvider, IFileSystem fileSystem)
         {
+            RuntimeConfig runtimeConfig = runtimeConfigProvider.GetConfig();
+            _runtimeConfigProvider = runtimeConfigProvider;
             _fileSystem = fileSystem;
-            _runtimeConfig = runtimeConfigProvider.GetConfig();
+            // Make a copy of the runtime entities to guarantee consistency
```

Comment:
"to guarantee consistency"
guarantee consistency of what? I know we may have spoken separately about this but no one else has this context and we may forget as time goes on.